e2e/test-component/scenarios/embedding-sdk/interactive-question-native.cy.spec.tsx

```diff
@@ -0,0 +1,54 @@
+import { createNativeQuestion, describeEE } from "e2e/support/helpers";
+import {
+  mockAuthProviderAndJwtSignIn,
+  mountInteractiveQuestion,
+  signInAsAdminAndEnableEmbeddingSdk,
+} from "e2e/support/helpers/component-testing-sdk";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import type { DatasetColumn } from "metabase-types/api";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
+
+describeEE("scenarios > embedding-sdk > interactive-question > native", () => {
+  beforeEach(() => {
+    signInAsAdminAndEnableEmbeddingSdk();
+  });
+
+  it("supports passing sql parameters to native questions", () => {
+    createNativeQuestion(
+      {
+        native: {
+          query: "SELECT * FROM orders WHERE {{ID}}",
+          "template-tags": {
+            ID: {
+              id: "6b8b10ef-0104-1047-1e1b-2492d5954322",
+              name: "ID",
+              "display-name": "ID",
+              type: "dimension",
+              dimension: ["field", ORDERS.ID, null],
+              "widget-type": "category",
+              default: null,
+            },
+          },
+        },
+      },
+      { wrapId: true },
+    );
+
+    cy.signOut();
+    mockAuthProviderAndJwtSignIn();
+    mountInteractiveQuestion({ initialSqlParameters: { ID: ORDERS_ID } });
+
+    // Should find a single row with the ID we passed in to the initial SQL parameters
+    cy.wait("@cardQuery").then(({ response }) => {
+      const { body } = response ?? {};
+
+      const idColumnIndex = body.data.cols.findIndex(
+        (column: DatasetColumn) => column.name === "ID",
+      );
+
+      expect(body.row_count).to.eq(1);
+      expect(body.data.rows[0][idColumnIndex]).to.eq(ORDERS_ID);
```

Comment:
As `ID` is a primary key in the sample database, we can safely assume that `{ID: ORDERS_ID}` should give us one row with the specified ID.

Reply:
I wonder if asserting the rendered UI is correct is better. To be fair, I'd feel more confident if we assert the UI, not the API response.

But I see the problem with going with the UI route is that I don't think we can see the amount of rows the API is returning.

Reply:
We should be able count the number of divs from the table root to get the row count too, but yes it's a little trickier to assert the UI. I just noticed we have some helpers in `assertTestData` though, so I can try to use that for assertions.

https://github.com/metabase/metabase/blob/6694870932830ff36964edd8b2c4555b81ddeda8/e2e/support/helpers/e2e-ui-elements-helpers.js#L310-L317

Reply:
well yeah, let's see how it goes, if we have a good argument to keep asserting the API then I'm also cool with that.

Reply:
@WiNloSt fixed in https://github.com/metabase/metabase/pull/50728/commits/c6ac1e7d96a54e08d8805e7a8a06d8373d828d3c, now it's asserting the table's UI instead.